application/frontend/controllers/MfaController.php

```diff
@@ -282,4 +290,45 @@ public function actionUpdate(int $id)
 
         return $mfa;
     }
+
+
+    /**
+     * @param int $mfaId
+     * @param int $webauthnId
+     * @return MfaWebauthn
+     * @throws BadRequestHttpException
+     * @throws NotFoundHttpException
+     * @throws ServerErrorHttpException
+     * @throws \Throwable
+     * @throws \yii\db\StaleObjectException
+     */
+    public function actionUpdateWebauthn(int $mfaId, int $webauthnId)
+    {
+        // check authorization
+        $this->getRequestedMfa($mfaId);
+
+
+        $webauthn = MfaWebauthn::findOne(['id' => $webauthnId, 'mfa_id' => $mfaId]);
+        if ($webauthn === null) {
+            throw new NotFoundHttpException(
+               "MfaWebauthn not found with id: $webauthnId and mfa_id: $mfaId",
+                1660232933
+            );
+        }
+
+        $label = \Yii::$app->request->getBodyParam('label');
+        $webauthn->label = $label?:"";
+
+        if ($webauthn->update() === false) {
+            \Yii::error([
+                'action' => 'update mfa webauthn',
+                'status' => 'error',
+                'error' => $webauthn->getFirstErrors(),
+                'mfa_id' => $webauthn->id,
+            ]);
+            throw new ServerErrorHttpException("Unable to update MfaWebauthn label", 1660233472);
+        }
```

Comment:
This is probably where we'll either need to first call `$webauthn->validate()` and return a 400 response if it fails, or else just return a 400 response for all unsuccessful calls to `$webauthn->update()`. I'm not sure whether those will all be bad-request failures, though. Some could be server errors, as you have it coded here currently.